src/test_report.py

```diff
@@ -64,23 +69,40 @@ def _get_group_data(self, root_node):
             'result': 'fail',
         })
         failures = [
-            node for node in failures if node['id'] != root_node['id']
+            node for node in failures if node['id']
         ]
+
         parent_path_len = len(root_node['path'])
         for node in failures:
+            if node['id'] == root_node['id']:
+                parent_path_len = len(checkout_node['path'])
             node['path'] = '.'.join(node['path'][parent_path_len:])
         return {'root': root_node, 'nodes': group_nodes, 'failures': failures}
 
+    def _get_groups(self, root_node):
```

Comment:
Hello @gctucker
To separate nodes based on groups, I have added a method to get all the distinct groups from child nodes.
Do we have a more efficient way to do this?